Makefile

```diff
@@ -671,6 +671,13 @@ test-e2e-local: ## Run "docker-build" rule then run e2e tests.
 	GINKGO_ARGS='$(LOCAL_GINKGO_ARGS)' \
 	test-e2e-run
 
+.PHONY: test-e2e-skip-build-and-push
+test-e2e-skip-build-and-push:
```

Comment:
Is this basically equivalent to removing the `make docker-build` from the `test-e2e` target? If so, I think adding some sort of toggle controlling whether or not the `docker-build` gets invoked during `make test-e2e` might be a slightly lighter-weight and more flexible way of accomplishing the same goal vs. adding a separate target.

Reply:
Yep, it's removing the docker-build and docker-push steps. Do we want to include conditional logic in the Makefile? Another flag we have is `LOCAL_ONLY`, and we handled the conditional logic in `./scripts/ci-e2e.sh` so I followed that pattern.

Reply:
My rule of thumb would generally be to keep the conditional logic as close as possible to the behavior it's modifying. So in this case I'd tend to put a condition tweaking flags on the e2e test invocation in the Makefile vs. in ci-e2e.sh where it's kind of one level higher. I think that would be a bit easier to follow (Makefile syntax depending). If that would go against an established pattern though I think following that is likely more valuable.